src/keyValueStore.js

```diff
@@ -57,6 +57,30 @@ class KeyValueStore {
     return true
   }
 
+  /**
+  * Set multiple values for multiple keys
+  *
+  * @param     {Array<String>}    keys                     the keys
+  * @param     {Array<String>}    values                   the values
+  * @return    {Boolean}                                  true if successful, throw error if not
+  */
+  async setMultiple (keys, values) {
+    throwIfNotEqualLenArrays(keys, values)
+    try {
+      await keys.reduce(async (previousPromise, nextKey, index) => {
+        await previousPromise
+        throwIfUndefined(nextKey, 'key')
+        this._requireLoad()
+        this._ensureConnected()
+        const timeStamp = new Date().getTime()
+        return this._db.put(nextKey, { value: values[index], timeStamp })
```

Comment:
Why not call `this.set` instead? Am I missing something?

Reply:
Weird! thought I left a comment about this. 

For some reason running the private store, calling `this.set` did not save the database entries. Not in the tests or in my actual application. As far as i could tell, it didn't have to do with race conditions, the `this` context, or anything else I investigated. I was (and still am) super confused, especially because it worked fine for the public store. [This commit](https://github.com/3box/3box-js/pull/409/commits/77ae568d775c9af23f2815823d9fa66e15c5fe47) solved the problem. Do you have any idea why that would be the case?

Reply:
I just verified this again in my actual application - returning `this.set` from within `setMultiple` causes private fields to return `null`. SO weird. Would love to know why this is happening.

returning `this._db.put` from inside `this.set` doesn't solve it. 

And old-school promisifying `this.set` doesn't do it either:

```js
set (key, value) {
    return new Promise(async (resolve, reject) => {
      throwIfUndefined(key, 'key')
      this._requireLoad()
      this._ensureConnected()
      const timeStamp = new Date().getTime()
      await this._db.put(key, { value, timeStamp })
      resolve(true)
    })
  }
```

Reply:
Oh, I think it's because `this` in that context might be the subclass and not the superclass! I.e. it uses privateStore `set` which tries to encrypt the entry again. Then when you get the entry it tries to decrypt it once, when it's encrypted two times.

Reply:
So yes, we should definitely not call `this.set` then. I think that's better though. We can move `this._requireLoad()` and `this._ensureConnected()` to be called only once in `setMultiple` 😃